js/components/graph/Graph.js

```diff
@@ -285,6 +293,45 @@ export class Graph extends React.Component {
     }
   }
 
+  updateEdgeMissingStatus = (edgeID, status) => {
+    this.setState(state => {
+      const temp = { ...state.edgeMissingStatus }
+      temp[edgeID] = status
+      return { edgeMissingStatus: temp }
+    })
+  }
+
+  /**
+   * Update the status of the Edge, this will take in a source, a target so it can be looked up
+   * We will loop through the edgesJSON and find use the source, target directly from edgeJSON
+   * the state will be stored and updated in edgeGroup object.
+   */
+  updateEdgeStatus = (source, target, status, edgeID) => {
+    const sourceNode = refLookUp(source, this)
```

Comment:
Move this logic (up to line 325) into the `Edge.updateStatus` function. The `Graph.updateEdgeStatus` function should only take in the `status` and `edgeID` to update.